library/system/src/lib/yast2/fs_snapshot.rb

```diff
@@ -0,0 +1,198 @@
+  class SnapperNotConfigured < StandardError
+    def initialize
+      super "Programming error: Snapper is not configured yet."
+    end
+  end
+
+  # Represents that does not exist a suitable 'pre' snapshot for a new 'post'
+  # snapshot.
+  class PreviousSnapshotNotFound < StandardError
+    def initialize
+      super "Previous snapshot was not found."
+    end
+  end
+
+  # Represents the fact that the snapshot could not be created.
+  class SnapshotCreationFailed < StandardError
+    def initialize
+      super "Filesystem snapshot could not be created."
+    end
+  end
+
+  # Class for managing filesystem snapshots. It's important to note that this
+  # class is intended to be used during installation/update so it uses the
+  # Snapper's CLI because the DBus interface is not available at that time.
+  class FsSnapshot
+    include Yast::Logger
+
+    FIND_CONFIG_CMD = "/usr/bin/snapper --no-dbus list-configs | grep \"^root \" >/dev/null"
+    CREATE_SNAPSHOT_CMD = "/usr/lib/snapper/installation-helper --step 5 --snapshot-type %s --description \"%s\""
+    LIST_SNAPSHOTS_CMD = "LANG=en_US.UTF-8 /usr/bin/snapper --no-dbus list"
+    VALID_LINE_REGEX = /\A\w+\s+\| \d+/
+
+    attr_reader :number, :snapshot_type, :previous_number, :timestamp, :user,
+      :cleanup_algo, :description
+
+    # Determines whether snapper is configured or not
+    #
+    # @return [true,false] true if it's configured; false otherwise.
+    def self.configured?
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), FIND_CONFIG_CMD)
+      log.info("Checking if Snapper is configured: \"#{FIND_CONFIG_CMD}\" returned: #{out}")
+      out["exit"] == 0
+    end
+
+    # Creates a new 'single' snapshot
+    #
+    # @param description [String] Snapshot's description.
+    # @return [FsSnapshot] The created snapshot.
+    #
+    # @see FsSnapshot.create
+    def self.create_single(description)
+      create(:single, description)
+    end
+
+    # Creates a new 'pre' snapshot
+    #
+    # @param description [String] Snapshot's description.
+    # @return [FsSnapshot] The created snapshot.
+    #
+    # @see FsSnapshot.create
+    def self.create_pre(description)
+      create(:pre, description)
+    end
+
+    # Creates a new 'post' snapshot
+    #
+    # Each 'post' snapshot corresponds with a 'pre' one. If the number of the
+    # 'pre' snapshot is not given, it will try to figure out that number.
+    #
+    # @param description     [String] Snapshot's description.
+    # @param previous_number [Fixnum] Number of the previous snapshot
+    # @return [FsSnapshot] The created snapshot.
+    #
+    # @see FsSnapshot.create
+    def self.create_post(description, previous_number = nil)
+      previous =
+        if previous_number.nil?
+          last = all.reverse.find { |s| s.snapshot_type == :pre || s.snapshot_type == :post }
+          (last && last.snapshot_type == :pre) ? last : nil
```

Comment:
On Tue, May 19, 2015 at 07:22:39AM -0700, Imobach González Sosa wrote:
> > +
> > +    # Creates a new 'post' snapshot
> > +    #
> > +    # Each 'post' snapshot corresponds with a 'pre' one. If the number of the
> > +    # 'pre' snapshot is not given, it will try to figure out that number.
> > +    #
> > +    # @param description     [String] Snapshot's description.
> > +    # @param previous_number [Fixnum] Number of the previous snapshot
> > +    # @return [FsSnapshot] The created snapshot.
> > +    #
> > +    # @see FsSnapshot.create
> > +    def self.create_post(description, previous_number = nil)
> > +      previous =
> > +        if previous_number.nil?
> > +          last = all.reverse.find { |s| s.snapshot_type == :pre || s.snapshot_type == :post }
> > +          (last && last.snapshot_type == :pre) ? last : nil
> 
> I guess that it could break also during upgrades/updates because zypper is ran, isn't it?

If snapper-zypp-plugin is installed then yes.